internal/core/window.rs

```diff
@@ -574,29 +577,45 @@ impl WindowInner {
                 crate::input::process_delayed_event(&window_adapter, mouse_input_state);
         }
 
-        if pressed_event {
-            self.had_popup_on_press.set(self.active_popup.borrow().is_some());
-        }
-
-        let close_policy = self.close_policy();
-        let mut mouse_inside_popup = false;
+        let close_policies = self.active_popups.borrow()
+            .iter()
+            .map(|item| (*item.0, item.1.close_policy))
+            .collect::<Vec<(u32, PopupClosePolicy)>>();
+        let mut clicked_popup_id = 0;
 
         mouse_input_state = if let Some(mut event) =
             crate::input::handle_mouse_grab(event, &window_adapter, &mut mouse_input_state)
         {
-            let (item_tree, offset) = if let Some(PopupWindow {
-                location: PopupWindowLocation::ChildWindow(coordinates),
-                component,
-                ..
-            }) = self.active_popup.borrow().as_ref()
-            {
-                let geom = ItemTreeRc::borrow_pin(component).as_ref().item_geometry(0);
+            let (item_tree, offset) = if !self.active_popups.borrow().is_empty() {
+                let active_popups = self.active_popups.borrow();
+
+                let clicked_popup = active_popups.iter().find(|popup| {
```

Comment:
Here i don't think we need to iterate over popup. Popup stacks and only the top one should be active at a given point. So I don't think the mouse handling needs any more changes.

Reply:
@ogoffart I was thinking all popups should be iterated because I'd expect all nested popups to close (or more specifically, context menus) when I click outside of the application. This seems to be the behaviour in most apps.

Reply:
I think I'll leave this to a different pull request because the current behaviour with a single popup is a bit odd as well.